modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/ProblemProviderImpl.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.vulcan.internal;
+
+import com.liferay.osgi.service.tracker.collections.map.ServiceReferenceMapperFactory;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMap;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMapFactory;
+import com.liferay.petra.reflect.GenericUtil;
+import com.liferay.portal.vulcan.problem.Problem;
+import com.liferay.portal.vulcan.problem.ProblemMapper;
+import com.liferay.portal.vulcan.problem.ProblemProvider;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+
+/**
+ * @author Alejandro Tardín
+ */
+@Component(service = ProblemProvider.class)
+public class ProblemProviderImpl implements ProblemProvider {
+
+	@Override
+	public Problem getProblem(Throwable throwable) {
+		while (throwable != null) {
+			Class<? extends Throwable> clazz = throwable.getClass();
```

Comment:
Mmm, what if there is a ProblemProvider of a superclass of the exception? I think we should not be considering it, right?

Reply:
Hmm I think we should also consider super classes... but for now this is enough for this use case. We can discuss it.

Reply:
ok, let's just have it in mind for the future